packages/source-map/tests/sourceMap.spec.ts

```diff
@@ -149,6 +149,162 @@ describe('sourceMap', () => {
 			],
 		]);
 
+		expect([...map.getGeneratedStartEnd(
+			`{{|data?.icon?.toString()}}`.indexOf('|'),
+			`{{data?.icon?.toString()|}}`.indexOf('|')
+		)].map(mapped => mapped.slice(0, 2))).toEqual([
+			[
+				0,
+				`(null as any ? ((null as any ? ((null as any ? (this.data)!.icon : undefined)!.toString : undefined))!() : undefined)`.length,
+			],
+		]);
+	});
+	test('Angular template with full ranges', () => {
```

Comment:
Added a test case without zero-length ranges and mappings with overlapping ranges.